specutils/spectra/spectral_axis.py

```diff
@@ -0,0 +1,98 @@
+import warnings
+
+import astropy.units as u
+from astropy.utils.decorators import lazyproperty
+import numpy as np
+
+from .spectral_coordinate import SpectralCoord
+
+
+__all__ = ['SpectralAxis']
+
+# We don't want to run doctests in the docstrings we inherit from Quantity
+__doctest_skip__ = ['SpectralAxis.*']
+
+class SpectralAxis(SpectralCoord):
+    """
+    Coordinate object representing spectral values corresponding to a specific
+    spectrum. Overloads SpectralCoord with additional information (currently
+    only bin edges).
+
+    Parameters
+    ----------
+    value : ndarray or `~astropy.units.Quantity` or `SpectralCoord`
+        Spectral axis data values.
+    unit : str or `~astropy.units.Unit`
+        Unit for the given data.
+    observer : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`, optional
+        The coordinate (position and velocity) of observer.
+    target : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`, optional
+        The coordinate (position and velocity) of observer.
+    radial_velocity : `~astropy.units.Quantity`, optional
+        The radial velocity of the target with respect to the observer.
+    redshift : float, optional
+        The redshift of the target with respect to the observer.
+    doppler_rest : `~astropy.units.Quantity`, optional
+        The rest value to use for velocity space transformations.
+    doppler_convention : str, optional
+        The convention to use when converting the spectral data to/from
+        velocity space.
+    bin_specificatoin: str, optional
+        Must be "edges" or "centers". Determines whether specified axis values
+        are interpreted as bin edges or bin centers. Defaults to "centers".
+    """
+
+    def __new__(cls, value, unit=None, observer=None, target=None,
+                 radial_velocity=None, redshift=None, doppler_rest=None,
+                 doppler_convention=None, bin_specification="centers",
+                 **kwargs):
+
+        # Convert to bin centers if bin edges were given, since SpectralCoord
+        # only accepts centers
+        if bin_specification == "edges":
+            bin_edges = value
+            value = SpectralAxis._centers_from_edges(value)
+
+        obj = super().__new__(cls, value, unit=unit, observer=observer,
+                              target=target, radial_velocity=radial_velocity,
+                              redshift=redshift, doppler_rest=doppler_rest,
+                              doppler_convention=doppler_convention, **kwargs)
```

Comment:
I think it's more appropriate to just keep the bundles init arguments buried in the `args` and `kwargs`, that way if we ever change something on the base `SpectralCoord` class, we won't need to also update the `SpectralAxis` class.

```suggestion
    def __new__(cls, value, *args, bin_specification="centers", **kwargs):

        # Convert to bin centers if bin edges were given, since SpectralCoord
        # only accepts centers
        if bin_specification == "edges":
            bin_edges = value
            value = SpectralAxis._centers_from_edges(value)

        obj = super().__new__(cls, value, *args, **kwargs)
```

Reply:
Good call - I figured there would be a couple things to improve with my subclassing, been a while since I was in OOP land.